### PR TITLE
Add report display options to admin page

### PR DESCRIPTION
**Pull request by michaelbish**
Add admin options to hide/show reports on home page

### CODE REVIEW

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs [hunk 1]
```diff
@@ -1,49 +1,72 @@
...
+        public ReportDetailsService(TableServiceClient tableServiceClient, IMemoryCache memoryCache, IHttpContextAccessor httpContextAccessor)
         {
             _tableServiceClient = tableServiceClient;
+            _memoryCache = memoryCache;
             _httpContextAccessor = httpContextAccessor;
         }
 
         public Task<IList<ReportDetail>> GetReportDetails() => RetrieveReportDetails();
-        public async Task<IList<ReportDetail>> GetReportDetailsForUser()
+        public async Task<IList<ReportDetail>> GetAllReportDetailsForUser()
```
Comment (emlyn-arkahna) [inline src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs:30]: Can we change this to GetAllReportDetails and remove the check for user groups please

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs [hunk 2]
```diff
@@ -1,49 +1,72 @@
...
         {
             var userGroups = _httpContextAccessor.HttpContext?.User.Claims
                 .Where(x => x.Type == "groups")
                 .Select(x => new Guid(x.Value))
                 .ToArray() ?? Array.Empty<Guid>();
 
-            return (await RetrieveReportDetails()).Where(x => x.Enabled && x.AadGroups.Any(a => userGroups.Contains(a))).ToList();
+            return (await RetrieveReportDetails()).Where(x => x.AadGroups.Any(a => userGroups.Contains(a))).ToList();
         }
+        public async Task<IList<ReportDetail>> GetEnabledReportDetailsForUser() => (await GetAllReportDetailsForUser()).Where(x => x.Enabled).ToList();
```
Comment (emlyn-arkahna) [inline src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs:39]: Can this be changed back to GetReportDetailsForUser and have the check for ad groups in it

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs [hunk 3]
```diff
@@ -1,49 +1,72 @@
...
-        public async Task<ReportDetail> GetReportForUser(Guid workspaceId, Guid reportId) => (await GetReportDetailsForUser()).FirstOrDefault(x => x.GroupId == workspaceId && x.ReportId == reportId);
+        public async Task<ReportDetail> GetReportForUser(Guid workspaceId, Guid reportId) => (await GetEnabledReportDetailsForUser()).FirstOrDefault(x => x.GroupId == workspaceId && x.ReportId == reportId);
 
-        private async Task<IList<ReportDetail>> RetrieveReportDetails()
-        {
+        public Task SaveReportDisplayDetails(IList<ReportDetail> reports, CancellationToken cancellationToken) => ModifyReportDetails(reports, cancellationToken);
+
+        private async Task<IList<ReportDetail>> RetrieveReportDetails() =>
+            await
+                _memoryCache.GetOrCreateAsync(CacheKey, async entry =>
```
Comment (emlyn-arkahna) [inline src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs:48]: Memory cache should not be here any more can you check you have merged the latest from 1.0

**Review by emlyn-arkahna — CHANGES_REQUESTED** [review]

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs (hunk 1 above)
Comment (michaelbish) [reply src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs]: Done

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs (hunk 3 above)
Comment (michaelbish) [reply src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs]: Done

File: src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs (hunk 2 above)
Comment (michaelbish) [reply src/PowerBiAuditApp.Client/Services/ReportDetailsService.cs]: Done

**Review by emlyn-arkahna — APPROVED** [review]
Approved

**Pull request merged**